crates/sui-core/src/authority.rs

```diff
@@ -2294,7 +2293,20 @@ impl AuthorityState {
         //
         // Only after CheckpointService::notify_checkpoint stores checkpoint in it's store we update checkpoint boundary
         if let Some((index, roots)) = self.database.last_checkpoint(round)? {
-            checkpoint_service.notify_checkpoint(index, roots, false)?;
+            let final_checkpoint_round = self.database.final_epoch_checkpoint()?;
```

Comment:
So there will be some period of gap between when we receive 2f+1 EndOfPublish and here. What happens in that peorid?

Reply:
We might have more narwhal transactions in the commit, and we will process them. Remaining certificates will be rejected, other messages like checkpoint signatures will be processed as usual